service/orchestrator/metrics_test.go

```diff
@@ -547,8 +706,19 @@ func TestService_GetMetricImplementation(t *testing.T) {
 		fields  fields
 		args    args
 		wantRes *assessment.MetricImplementation
-		wantErr bool
+		wantErr assert.ErrorAssertionFunc
 	}{
+		{
+			name:    "Invalid input",
+			wantRes: nil,
+			wantErr: func(tt assert.TestingT, err error, i ...interface{}) bool {
+				gotStatus, ok := status.FromError(err)
```

Comment:
minor: Since this seems to repeat a lot of times, would it make sense to build a function that we could directly use as as `assert.ErrorAssertionFunc? Something like:
```go
func wantStatusCode(code codes.Code) assert.ErrorAssertionFunc {
	return func(t assert.TestingT, err error, i ...interface{}) bool {
		gotStatus, ok := status.FromError(err)
		if !ok {
			return false
		}
		return assert.Equal(t, gotStatus.Code(), code)
	}
}
```

this can then be used directly:
```go
...
wantErr: wantStatusCode(codes.InvalidArgument)
```